controllers/storagecluster/reconcile.go

```diff
@@ -241,8 +241,11 @@ func (r *StorageClusterReconciler) validateStorageClusterSpec(instance *ocsv1.St
 		return err
 	}
 
-	if instance.Spec.ExternalStorage.Enable && instance.Spec.ExternalStorage.StorageProviderKind == ocsv1.KindOCS && instance.Spec.ExternalStorage.ConnectionString == "" {
-		err := fmt.Errorf("spec.externalStorage.connectionString can not be empty when spec.externalStorage.storageProviderKind is set to ocs")
+	if instance.Spec.ExternalStorage.Enable && instance.Spec.ExternalStorage.StorageProviderKind == ocsv1.KindOCS &&
+		(instance.Spec.ExternalStorage.ConnectionString == "" || instance.Spec.ExternalStorage.RequestedCapacity == nil) {
```

Comment:
You can use the `isExternalOCSProvider` helper to make the condition more clear

```suggestion
	if isExternalOCSProvider(instance) && (instance.Spec.ExternalStorage.ConnectionString == "" || instance.Spec.ExternalStorage.RequestedCapacity == nil) {
```

Reply:
done